active/NavigationView/HierarchicalNav/HierarchicalNavigationView.md

```diff
@@ -0,0 +1,272 @@
+            if (nvib != null)
+            {
+                var name = "Last Collapsed: " + nvib.Content;
+                CollapsedItemLabel.Text = name;
+            }
+            else
+            {
+                CollapsedItemLabel.Text = "Last Collapsed: ERROR - No container returned!";
+            }
+        }
+```
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples
+section.  For example, see the Remarks for the MediaPlayerElement 
+(https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.MediaPlayerElement#remarks). -->
+
+
+# API Notes
+<!-- Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. -->
+
+
+# API Details
+```c++
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass NavigationViewExpandingEventArgs
+{
+    NavigationViewItemBase ExpandingItemContainer { get; };
+}
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass NavigationViewCollapsedEventArgs
+{
+    NavigationViewItemBase CollapsedItemContainer { get; };
+}
+
+[WUXC_VERSION_RS3]
+[webhosthidden]
+[WUXC_INTERFACE_NAME("INavigationView", f209ce15-391a-42ca-9fc6-f79da65aca32)]
+[WUXC_STATIC_NAME("INavigationViewStatics", 363a86c7-72da-4420-b871-15d9d0d45756)]
+[WUXC_CONSTRUCTOR_NAME("INavigationViewFactory", e50687c1-b7c2-4975-ad7a-5f4fe6a514c9)]
+[MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+[MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]
+unsealed runtimeclass NavigationView : Windows.UI.Xaml.Controls.ContentControl
+{
+...
+    [WUXC_VERSION_PREVIEW]
+    {
+        event Windows.Foundation.TypedEventHandler<NavigationView, NavigationViewExpandingEventArgs> Expanding;
+        event Windows.Foundation.TypedEventHandler<NavigationView, NavigationViewCollapsedEventArgs> Collapsed;
+
+        void Expand(NavigationViewItem item);
+        void Collapse(NavigationViewItem item);
+    }
+}
+
+[WUXC_VERSION_RS3]
+[webhosthidden]
+[WUXC_INTERFACE_NAME("INavigationViewItemBase", edf04eb1-37d1-471f-8570-3829ee5b2bc6)]
+[WUXC_CONSTRUCTOR_NAME("INavigationViewItemBaseFactory", eb014cef-7890-4ebb-8245-02e8510f321d)]
+[default_interface]
+unsealed runtimeclass NavigationViewItemBase : Windows.UI.Xaml.Controls.ContentControl
+{
+    [WUXC_VERSION_PREVIEW]
+    {
+        Boolean IsSelected{ get; set; };
+        static Windows.UI.Xaml.DependencyProperty IsSelectedProperty { get; };
+    }
+}
+
+[WUXC_VERSION_RS3]
+[webhosthidden]
+[WUXC_INTERFACE_NAME("INavigationViewItem", 8614be0f-b7b6-4851-960a-f5e3f69f624a)]
+[WUXC_STATIC_NAME("INavigationViewItemStatics", 803c0081-fda5-4b90-aace-3f2306dbe5c4)]
+[WUXC_CONSTRUCTOR_NAME("INavigationViewItemFactory", 973bdb4a-7e08-4f76-923c-f12bd685e86e)]
+unsealed runtimeclass NavigationViewItem : NavigationViewItemBase
+{
+...
+    [WUXC_VERSION_PREVIEW]
+    {
+        [MUX_DEFAULT_VALUE("false")]
+        Boolean IsExpanded{ get; set; };
```

Comment:
This is silently ignored if this has no children?

Reply:
I think it should return false in all circumstances other than when it's a NavigationViewItem with MenuItems and those items are visible because it's expanded. (this property will also be available on NavigationViewItemSeparator and NavigationViewItemHeader, which inherit from NVIB.)

Reply:
It's not unheard of, but it's unusual to set something true and have it read back false. Having this property on the separator/header is odd, and doesn't seem necessary since this property isn't on the Base?

Reply:
Oh, I misread the parent class, headers and separators won't have IsExpanded.

So yes, I expect setting IsExpanded to true on a leaf node to get ignored.

Reply:
I would expect setting IsExpanded to true/false will expand or collapse the item if it contains any children, do nothing otherwise. The last written value of the property should be what is read back.